### PR TITLE
Add flash attention support for FP8 Mistral

### PR DESCRIPTION
**Pull request by schoi-habana**
--use_flash_attention was not used for FP8 Mistral due to the accuracy issue (https://github.com/huggingface/optimum-habana/pull/985). Since the accuracy issue is now gone in 1.17.0, this PR refactor the mistral modeling script to enable flash attention for FP8 flow. Also the GaudiMistralAttentionLongSequence class is removed because the model can serve seq_len==max_position_embedding and performs better without it.

For the case with  --batch_size 7 --max_new_tokens 512 --max_input_tokens 32000 

1. Original code without flash attention:
Throughput (including tokenization) = 42.07969811168232 tokens/second
Number of HPU graphs                = 121
Memory allocated                    = 37.51 GB
Max memory allocated                = 94.31 GB
Total memory available              = 94.62 GB
Graph compilation duration          = 285.1875329967588 seconds
Time to first token                       = 77804.80473767966ms

2.  This PR with flash attention:
Throughput (including tokenization) = 63.06745994995644 tokens/second
Number of HPU graphs                = 85
Memory allocated                    = 37.52 GB
Max memory allocated                = 94.0 GB
Total memory available              = 94.62 GB
Graph compilation duration          = 277.9795082975179 seconds
Time to first token = 49177.13945917785ms

command: QUANT_CONFIG=./quantization_config/maxabs_quant.json python run_generation.py --model_name_or_path /root/tf/test/mistral/ --attn_softmax_bf16 --use_hpu_graphs --trim_logits --use_kv_cache --reuse_cache --bf16 --batch_size 7 --max_new_tokens 512 --max_input_tokens 32000 --limit_hpu_graphs --use_flash_attention --flash_attention_recompute 

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

### CODE REVIEW

**Review by mandy-li — APPROVED** [review]
LGTM

File: optimum/habana/transformers/models/mistral/modeling_mistral.py [hunk 1]
```diff
@@ -386,28 +389,35 @@ def forward(
...
-                            query_states, key_states, value_states, None, 0.0, True, None
-                        )
+                if (
+                    not self.training
+                    and q_len == key_states.size(-2)
+                    and q_len > 8192
+                    and os.getenv("QUANT_CONFIG", "")
+                ):
+                    htcore.mark_step()
+                    attn_output = self.fused_sdpa_long(
```
Comment (jiminha) [inline optimum/habana/transformers/models/mistral/modeling_mistral.py:399]: Do we still need this special handling for the longer sequence? Last time when I tried as long as the use_flash_attention works, that showed better performance. wondering if we still need this special handling for the longer q_len.

**Review by jiminha — APPROVED** [review]
